### PR TITLE
fix setting of location attribute in init

### PR DESCRIPTION
**Pull request by crdunwel**
Also added convenience method for setting it. See http://docs.opencivicdata.org/en/latest/proposals/0004.html#location

### CODE REVIEW

File: pupa/scrape/event.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def __init__(self, name, start_time, timezone, location, *,
         self.description = description
         self.status = status
         self.classification = classification
-        self.location = {"name": location, "note": "", "coordinates": None}
+        self.location = location
```
Comment (paultag) [inline pupa/scrape/event.py]:
```diff
@@ -62,7 +62,7 @@ class Event(BaseModel, SourceMixin, AssociatedLinkMixin, LinkMixin):
self.description = description
self.status = status
self.classification = classification
-        self.location = {"name": location, "note": "", "coordinates": None}
+        self.location = location
```

Why this change? Seems like that forces folks to create this dict? I'd either
set it to None and handle that elsewhere or keep the full dict here. Pehaps
changing the name of that kwarg to location_name would be better?

Comment (crdunwel) [inline pupa/scrape/event.py]: The change is because the variable name `location` is a little confusing for initialization. As you can see currently, the variable is setting just the `name` key for the location attribute on the Event model. I agree that changing the variable name to `location_name` would be better for sake of clarity and convenience since note + coordinates may often be blank anyways. Will submit new PR with method for setting note + coordinates attributes.

Comment (paultag) [inline pupa/scrape/event.py]: `location_name` SGTM; also tests fail on this changeset

On Mon, May 18, 2015 at 2:30 PM, Clayton Dunwell <notifications@github.com>
wrote:

> In pupa/scrape/event.py
> <https://github.com/opencivicdata/pupa/pull/180#discussion_r30534604>:
>
> > @@ -62,7 +62,7 @@ def __init__(self, name, start_time, timezone, location, *,
> >          self.description = description
> >          self.status = status
> >          self.classification = classification
> > -        self.location = {"name": location, "note": "", "coordinates": None}
> > +        self.location = location
>
> The change is because the variable name location is a little confusing
> for initialization. As you can see currently, the variable is setting just
> the name key for the location attribute on the Event model. I agree that
> changing the variable name to location_name would be better for sake of
> clarity and convenience since note + coordinates may often be blank
> anyways. Will submit new PR with method for setting note + coordinates
> attributes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencivicdata/pupa/pull/180/files#r30534604>.
>



-- 
Paul Tagliamonte
Software Developer | Sunlight Foundation

**Comment by paultag** [comment]
There's a merge conflict in `pupa/scrape/event.py` now (I think from the last PR)

**Comment by crdunwel** [comment]
I can just submit a new PR? Will be what was discussed above.

**Comment by paultag** [comment]
Sure!

On Tue, May 19, 2015 at 10:59 AM, Clayton Dunwell <notifications@github.com>
wrote:

> I can just submit a new PR? Will be what was discussed above.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencivicdata/pupa/pull/180#issuecomment-103537721>.
>



-- 
Paul Tagliamonte
Software Developer | Sunlight Foundation
